packages/headless/src/controllers/search-box/headless-search-box.ts

```diff
@@ -172,7 +173,10 @@ export function buildSearchBox(
 
   const performSearch = (analytics: SearchAction) => {
     const {enableQuerySyntax} = options;
-
+    const {clearFiltersOnNewQuery} = options;
+    if (clearFiltersOnNewQuery) {
+      dispatch(deselectAllFacets());
```

Comment:
Following on my earlier comment,

I think this PR can be simplified to just adding this line, with no option to allow changing the behavior.

I see facets as dynamic filters, that should reset when a search is performed.
They are different from tabs, aka static filters, that remain selected across multiple searches.

Reply:
What if a customer doesn't want to have that automatic clear, they would have to save them somehow? I'm sure there's a use case in support for keeping those dynamic filters up.

Reply:
https://discuss.coveo.com/t/clear-facet-selection-for-a-new-query-or-not/2347 

https://coveord.atlassian.net/wiki/spaces/UX/pages/723124387/Clearing+Filters+on+New+Query

That's actually a pretty big subject 😂

Reply:
If we agree keeping active filters is not good UX, and there is no major site using such a behavior (amazon, google, youtube, ebay), then a customer who wants to go against the flow could still do that by dispatching actions directly. The controllers should not make it easy to implement a bad UX.


If there is a major site that is keeping filters, then an option would make sense, but every site I've checked is clearing them. Atm, I see the jira as a bug in the search-box controller. I would change my mind if there is a site out there keeping filters,

Reply:
> if there is an example out there keeping filters

Unfortunately, most of our customers implementation 😄 
Okay, let's remove the option in atomic & headless and see what happens!

Reply:
so I should just keep the dispatch(deselectAllFacets()) line and remove the clearFiltersOnNewQuery variable altogether?

Reply:
😕  Yes, it could be helpful to update the submit method documentation to make the behaviour clear.

Reply:
Ok, I will do that